climada/entity/exposures/crop_production.py

```diff
@@ -256,8 +300,11 @@ def set_from_single_run(self, input_dir=INPUT_DIR, filename=None, hist_mean=HIST
         # The exposure [t/y] is computed per grid cell as the product of the area covered
         # by a crop [ha] and its yield [t/ha/y]
         self['value'] = np.squeeze(area_crop[irr[0]]*hist_mean_dict[irr[0]][idx_mean])
+        self['value'] = np.nan_to_num(self.value) # replace NaN by 0.0
         for irr_val in irr[1:]: # add other irrigation types if irr=combined
-            self['value'] += np.squeeze(area_crop[irr_val]*hist_mean_dict[irr_val][idx_mean])
```

Comment:
That is not a very clear use. Why would this be needed iif `irr = combined`? What happens if e.g. `irr = ['firr', 'combined', None]` ?

Reply:
There are three options for irr:
_firr_, _noirr_, or _combined_. Only if `irr=='combined'`, the variable is changed to the list `['firr', 'noirr']`.

`irr = ['firr', 'combined', None]`  is not a valid option, if users provide this as input there will be a ValueError because there is no `area_crop['combined']`

Reply:
That is suboptimal. A variable should not change type during the code. It should not go from str to list.

Reply:
agreed, I will disentangle this by defining an extra list variable `irr_types`